src/rpc/mining.cpp

```diff
@@ -814,20 +812,19 @@ static RPCHelpMan getblocktemplate()
 
         // Create new block
         CScript scriptDummy = CScript() << OP_TRUE;
-        pblocktemplate = miner.createNewBlock(scriptDummy);
-        if (!pblocktemplate) {
-            throw JSONRPCError(RPC_OUT_OF_MEMORY, "Out of memory");
-        }
+        block_template = miner.createNewBlock(scriptDummy);
+        CHECK_NONFATAL(block_template);
+
 
         // Need to update only after we know createNewBlock succeeded
         pindexPrev = pindexPrevNew;
     }
     CHECK_NONFATAL(pindexPrev);
-    CBlock* pblock = &pblocktemplate->block; // pointer for convenience
+    CBlock block{block_template->getBlock()};
```

Comment:
Likely a dumb question, but is switching from a `CBlock` pointer to using `getBlock` which returns a value directly going to have much of a performance impact (lots of copying)?

Reply:
> Likely a dumb question, but is switching from a `CBlock` pointer to using `getBlock` which returns a value directly going to have much of a performance impact (lots of copying)?

Definitely not a dumb question, this changes result in an extra copy (only a move is happening on this line, but there is a copy inside the `getBlock()` function, but a copy is also removed later on due to adding std::move in `GenerateBlock`, so the net number of copies should be the same.

Reply:
Marking as unresolved so we don't lose track of it.

Reply:
Sorry, I didn't look at where GenerateBlock function was being called, I just saw one copy being added and another being removed and assumed net number of copies was the same. Agree that cost of copying should not be significant relative to other things.